src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java

```diff
@@ -242,7 +242,7 @@ private void validateIndexTarget(TableMetadata table, IndexMetadata.Kind kind, I
             throw ire(ONLY_PARTITION_KEY, column);
 
         if (column.type.isFrozenCollection() && target.type != Type.FULL)
-            throw ire(CREATE_ON_FROZEN_COLUMN, target.type, column, column);
+            throw ire(CREATE_ON_FROZEN_COLUMN, target.type, column, column.name.toCQLString());
```

Comment:
not for this patch: honestly, when do we want toString to not be valid CQL?  this is a common issue...

Reply:
It's just a matter of the quotes, right?

Reply:
for the most part yah..